design-documents/asynchronous-import/base-extension.md

```diff
@@ -207,70 +166,88 @@ Will return list of Source that was uploaded before.
 }
 ```
 
-## Start File Import Endpoint
+### Base CSV Parser
+
+As in first iteration we will support CSV format only, will be implemented CSV reader.
+
+#### Problem
+In case of heavy import data, it may require a lot of memory resources to hold all information from the file in the memory.
+
+#### Solution
+Will be implemented CSV reader that will read source partially, process information and then proceed with next part.
+
+### Asynchronous web endpoints
+As we are moving in service isolation approach, we will support 2 ways of communicating between Asynchronous Import and Magento.
+
+* Service installed with monolith: will be user MFQF and Direct usage of Service Contracts
+* Service installed independently: HTTP communication by using Bulk API
+
+
+## Start Import Endpoint
+
 ### Main endpoint
 
 Current Endpoint starts import process based on FileID. Where Module will read file, split it into message and send to Async API
 
-POST  `/V1/import/start/{uuid}`
-
-`type` - its an import type like: catalog_product, catalog_category, customer, order ... etc...
+POST  `/V1/imports`
 
-Start File Import
+Start Import
 
 ```
 {
-    "importConfig": {
+    "import": {
+        "uuid": "",
+        "source_uuid": "",
         "import_type": "catalog_product",
-        "import_strategy": "add_update, delete, replace",
         "validation_strategy": "string",
         "allowed_error_count": 0,
         "import_image_archive": "string",
-        "import_images_file_dir": "string",
-        "mapping": [
-              {
-                  "name": "string",
-                  "source_path": "string",
-                  "target_path": "string",
-                  "target_value": "mixed",
-                  "processing_rules": [
-                        {
-                              "sort": int,
-                              "function": "string",
-                              "args": "array"
-                        }
-                  ]
-              }
-        ]
-    }
+        "import_images_file_dir": "string"
+    },
+    "convertingRules": [
```

Comment:
1- Mapping is a commonly used term for Data Mapping, 
2- Not sure, if I can understand how to call this API, it is not clear the intend of the parameters to be honest.
3- What is the use-case of sorting while importing? usually sorting of any kind doesn't happen during data importing.

Reply:
Thats fine, its not so easy from first glance :-)
convertingRules are defining how values for import can be changed to archive our goals. 
Same time its not require param.
Out of box, without this param we are supporting standard Magento CSV format.
But imaging real world where each system export their own formats. This mean we have to provide possibility to configure import source so, that customer can import any csv data structure he wants.
Here are coming convertingRules. 

So we have in the code some sub-module: RulesConverter.
Module takes all data we giving him (in this case read from the source file).
Then converter apply rules for this data.
Example: In import Status == Yes/No, we have a rule that convert this to 0/1 .... and so on for all required NOT standard Magento fields. 
And it returns already changed input data, but those data are ready for import to Magento.

Sort order is required, cause you may have multiple manipulation with same field, so order of execution have to be correct.

Reply:
thanks for clarification,